test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -766,4 +766,38 @@ describe('Loader hooks', { concurrency: true }, () => {
     assert.strictEqual(code, 0);
     assert.strictEqual(signal, null);
   });
+
+  it('should `deregister` properly', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--input-type=module',
+      '--eval',
+      `
+        import {register, deregister} from 'node:module';
+        const id = register(
+          ${JSON.stringify(fixtures.fileURL('/es-module-loaders/hooks-meta.mjs'))}
+        );
+
+        await import('node:os');
+        await import('node:os');
+
+        console.log('deregister', deregister(id));
+
+        await import('node:os');
+      `,
+    ]);
+
+    const lines = stdout.split('\n');
+
+    assert.strictEqual(lines.length, 5);
+    assert.strictEqual(lines[0], 'hooks initialize');
+    assert.strictEqual(lines[1], 'resolve passthru');
+    assert.strictEqual(lines[2], 'resolve passthru');
+    assert.strictEqual(lines[3], 'deregister true');
+    assert.strictEqual(lines[4], '');
+
+    assert.strictEqual(stderr, '');
```

Comment:
```suggestion
    assert.strictEqual(stderr, '');
    assert.deepStrictEqual(stdout.split('\n'), [ 'hooks initialize',
                                                 'resolve passthru',
                                                 'resolve passthru',
                                                 'deregister true',
                                                 '' ]);
```

Reply:
i'm good with comparing to an array, but please don't do this awful whitespace